GVFS/GVFS.Common/Prefetch/BlobPrefetcher.cs

```diff
@@ -513,9 +479,59 @@ protected void DownloadMissingCommit(string commitSha, GitObjects gitObjects)
             }
         }
 
+        private static bool TryLoadFileOrFolderList(Enlistment enlistment, string valueString, string listFileName, bool isFolder, List<string> output, Func<string, string> elementValidationFunction, out string error)
+        {
+            output.AddRange(
+                valueString.Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries)
+                .Select(path => BlobPrefetcher.ToAbsolutePath(enlistment, path, isFolder: isFolder)));
+
+            if (!string.IsNullOrWhiteSpace(listFileName))
+            {
+                IEnumerable<string> linesFromFile = null;
+                if (File.Exists(listFileName))
+                {
+                    linesFromFile = File.ReadAllLines(listFileName);
+                }
+                else if (listFileName == StdInFileName)
+            {
+                    linesFromFile = Console.In
+                        .ReadToEnd()

```

Comment:
`ReadToEnd().Split()` seems fraught with performance problems. Please replace with a `Console.ReadLine()` loop. Saves you the trouble with `Environment.NewLine`, too.